saltswap/tests/test_recorder.py

```diff
@@ -85,3 +87,5 @@ def test_write_read(self):
         for var, res in zip(sample_variables, saved_results):
             assert np.all(opened_ncfile.groups['Sample state data'][var][:] == res)
 
+        remove(filename)
```

Comment:
Might want to make sure that the file is closed before calling remove. Does this run without error?